cloudcix_primitives/network_secondary_lxd.py

```diff
@@ -0,0 +1,225 @@
+
+__all__ = [
+    'build',
+    'read',
+    'scrub',
+]
+
+def build(
+    endpoint_url: str,
+    project: str,
+    container_name: str,
+    network_interface_name: str,
+    verify_lxd_certs: bool = True,
+) -> Tuple[bool, str]:
+    """
+    Attach a secondary network interface to an LXD container.
+    """
+    # Use network_interface_name as the device_name
+    device_name = network_interface_name
+    
+    # Define the messages
+    messages = {
+        1000: f'Successfully attached {network_interface_name} to container {container_name} on {endpoint_url}',
+        3021: f'Failed to connect to {endpoint_url} for network or container operations',
+        3022: f'Failed to retrieve networks from {endpoint_url}. Payload exited with status ',
+        3023: f'Failed to retrieve container {container_name}. Payload exited with status ',
+        3024: f'Network device {device_name} already exists on container {container_name}.',
+        3025: f'Network {network_interface_name} does not exist on {endpoint_url}.',
+        3026: f'Failed to create network device: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+        
+        # Check if the network exists
+        ret = rcc.run(cli='networks.all')
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+        
+        networks = ret['payload_message']
+        fmt.add_successful('networks.all', ret)
+        
+        if not any(network.name == network_interface_name for network in networks):
+            return False, f"{prefix+5}: {messages[prefix+5]}", fmt.successful_payloads
+        
+        # Get the container
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+3}: {messages[prefix+3]}"), fmt.successful_payloads
+    
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+        
+        # Check if the device already exists
+        devices = instance.devices
+        if device_name in devices:
+            return False, f"{prefix+4}: {messages[prefix+4]}", fmt.successful_payloads
+        
+        # Create device configuration
+        device_config = {
```

Comment:
https://github.com/CloudCIX/primitives/blob/a6762a3c423ba27e96e151ecf60ca39c428b019b/cloudcix_primitives/lxd.py#L192

There is a difference in the properties for secondary interfaces here and during build. Does build need to be revisited? https://github.com/CloudCIX/primitives/blob/a6762a3c423ba27e96e151ecf60ca39c428b019b/cloudcix_primitives/lxd.py#L192

Also, is this primitive for managing "secondary interface devices" but the IP configuration will occur with "network_primary_lxd" similar to build taking in the information in `network_config`